src/main/java/org/prebid/server/manager/AdminManager.java

```diff
@@ -0,0 +1,87 @@
+package org.prebid.server.manager;
+
+import lombok.AllArgsConstructor;
+
+import java.time.Instant;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiConsumer;
+
+public class AdminManager {
+
+    public static final String ADMIN_COUNTER_KEY = "admin_counter";
+    public static final String ADMIN_TIME_KEY = "admin_time";
+
+    private ConcurrentHashMap<String, Rule<?, ?>> actionMap;
+
+    public AdminManager() {
+        actionMap = new ConcurrentHashMap<>();
+    }
+
+    public void setupByCounter(String key, Integer amount, BiConsumer<?, ?> action, BiConsumer<?, ?> onFinish) {
+        actionMap.put(key, new CounterRule(action, onFinish, amount));
+    }
+
+    public void setupByTime(String key, long timeMillis, BiConsumer<?, ?> action, BiConsumer<?, ?> onFinish) {
+        actionMap.put(key, new TimeRule(action, onFinish, timeMillis));
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T, U> void accept(String key, T t, U u) {
+        final Rule<T, U> rule = (Rule<T, U>) actionMap.get(key);
+        rule.applyRule().accept(t, u);
+    }
+
+    public boolean contains(String key) {
+        return actionMap.containsKey(key);
+    }
+
+    @AllArgsConstructor
+    private abstract static class Rule<T, U> {
+
+        protected BiConsumer<T, U> onRun;
+
+        protected BiConsumer<T, U> onFinish;
+
+        abstract BiConsumer<T, U> applyRule();
+    }
+
+    private static class TimeRule<T, U> extends Rule<T, U> {
+
+        private Long time;
+
+        TimeRule(BiConsumer<T, U> onRun, BiConsumer<T, U> onFinish, Long timeMillis) {
+            super(onRun, onFinish);
+            this.onFinish = onFinish;
+            this.time = Instant.now().toEpochMilli() + timeMillis;
```

Comment:
Actually `time` could be `Instant` as well. This will give an opportunity to use nice `isBefore` or `isAfter` methods. Also take note of `plus` method for getting another instant.
Another field for improvement: instead of `Instance.now()` use `Instance.now(Clock)` - this will ease testing a lot and will allow to get rid of `Thread.sleep()` (not 100% sure about that though :)).